core/dbt/tree_sitter_jinja/extractor.py

```diff
@@ -0,0 +1,292 @@
+from dataclasses import dataclass
+from functools import reduce
+from itertools import dropwhile
+from tree_sitter import Parser  # type: ignore
+from tree_sitter_jinja2 import JINJA2_LANGUAGE  # type: ignore
+
+
+# global values
+parser = Parser()
+parser.set_language(JINJA2_LANGUAGE)
+
+
+@dataclass
+class ParseFailure(Exception):
+    msg: str
+
+
+@dataclass
+class TypeCheckFailure(Exception):
+    msg: str
+
+
+def named_children(node):
+    return list(filter(lambda x: x.is_named, node.children))
+
+
+def text_from_node(source_bytes, node):
+    return source_bytes[node.start_byte:node.end_byte].decode('utf8')
+
+
+def strip_quotes(text):
+    if text:
+        return text[1:-1]
```

Comment:
(optional) might be able to use python's built-in [strip](https://docs.python.org/3/library/stdtypes.html#str.strip) here!

Reply:
Just checked this out and I think it's better to use the explicit indexing because of the following edge case which removes too many characters:
```
>>> x = '"nested "quotes""'
>>> x.strip('"')
'nested "quotes'
```